_posts/2021-03-23-connector-build.md

```diff
@@ -0,0 +1,232 @@
+---
+layout: post
+title:  "How to build your own KafkaConnect image via custom resource"
+date: 2021-03-23
+author: jakub_stejskal
+---
+In one of [the older blog post](https://strimzi.io/blog/2020/05/07/camel-kafka-connectors/) we shown you how easy is to integrate Camel with Strimzi via KafkaConnect custom resource.
+That approach had one limitation - you had to build your own KafkaConnect image and use it in custom resource.
+However, this step is not needed anymore, thanks to feature which we introduced in Strimzi 0.22 to allow users to build custom KafkaConnect images directly by Strimzi.
+
+### How does it work
+Strimzi uses two different ways how to build the image based on the underlying kubernetes cluster:
+* In case you use pure Kubernetes, the image will be built by [Kaniko](https://github.com/GoogleContainerTools/kaniko)
```

Comment:
On the Strimzi blog I would maybe not refer to _pure_ Kubernetes. That is more a Red Hat thing calling non-OpenShift distros _pure Kubernetes_ I think.

```suggestion
* In case you use Kubernetes, the image will be built by [Kaniko](https://github.com/GoogleContainerTools/kaniko)
```

Maybe we can also give one or two sentences describing Kaniko? Many people might not know it - and since we built on top of them it would be nice to give them some nice reference. We can probably just copy the intro part from their GitHub:

_kaniko is a tool to build container images from a Dockerfile, inside a container or Kubernetes cluster. kaniko doesn't depend on a Docker daemon and executes each command within a Dockerfile completely in userspace. This enables building container images in environments that can't easily or securely run a Docker daemon, such as a standard Kubernetes cluster._

Maybe we should also refer to it as `kaniko` and not `Kaniko`? They seem to use lowercase on their GitHub but uppercase `K` in the logo.

Reply:
ok, I quoted this description in text and used `kaniko`